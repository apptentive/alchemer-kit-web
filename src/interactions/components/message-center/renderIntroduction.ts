/**
 * Renders the Message Center intro (automated_message).
 *
 * @param {string} introText - The content to use for the introduction
 * @returns {HTMLDivElement} - The intro DOM Node.
 */
const renderIntroduction = (introText?: string) => {
  const introElement = document.createElement('div');
  introElement.classList.add('apptentive-intro');

  if (introText) {
    const introTextElement = document.createElement('p');
    introTextElement.classList.add('apptentive-typography__lead');
    introTextElement.textContent = introText;

    introElement.append(introTextElement);
  }

  return introElement;
};

export default renderIntroduction;
