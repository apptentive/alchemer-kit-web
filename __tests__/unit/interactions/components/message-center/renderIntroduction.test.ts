import renderIntroduction from '../../../../../src/interactions/components/message-center/renderIntroduction';

const _introText = 'Introduction';

describe('renderIntroduction', () => {
  test('properly renders introduction', () => {
    const output = renderIntroduction(_introText);

    expect(output).toMatchSnapshot();
  });

  test('properly renders introduction with no text', () => {
    const output = renderIntroduction();

    expect(output).toMatchSnapshot();
  });
});
