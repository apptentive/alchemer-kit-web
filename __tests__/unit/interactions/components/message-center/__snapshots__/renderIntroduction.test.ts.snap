// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderIntroduction properly renders introduction 1`] = `
<div
  class="apptentive-intro"
>
  <p
    class="apptentive-typography__lead"
  >
    Introduction
  </p>
</div>
`;

exports[`renderIntroduction properly renders introduction with no text 1`] = `
<div
  class="apptentive-intro"
/>
`;
